packages/@lwc/integration-karma/test/template/attribute-class/index.spec.js

```diff
@@ -77,4 +77,44 @@ describe('dynamic class attribute', () => {
             expect(target.className).toBe('bar baz');
         });
     });
+
+    describe('updating with null/undefined/empty string', () => {
+        function testRenderClassAttribute(type, value, expectedValue) {
+            it(`renders the class attribute for ${type}`, () => {
+                const elm = createElement('x-dynamic', { is: Dynamic });
+                elm.dynamicClass = value;
+                document.body.appendChild(elm);
+
+                expect(elm.shadowRoot.querySelector('div').getAttribute('class')).toBe(
+                    expectedValue
+                );
+            });
+        }
+
+        testRenderClassAttribute('null', null, null);
+        testRenderClassAttribute('undefined', undefined, null);
+        testRenderClassAttribute('empty string', '', null);
+        testRenderClassAttribute('class string', 'my-class', 'my-class');
+
+        function testUpdateClassAttribute(type, value, expectedValue) {
+            it(`updates the class attribute for ${type}`, async () => {
+                const elm = createElement('x-dynamic', { is: Dynamic });
+                elm.dynamicClass = 'my-class';
+                document.body.appendChild(elm);
+
+                expect(elm.shadowRoot.querySelector('div').getAttribute('class')).toBe('my-class');
+
+                elm.dynamicClass = value;
+                await Promise.resolve();
+                expect(elm.shadowRoot.querySelector('div').getAttribute('class')).toBe(
+                    expectedValue
+                );
+            });
+        }
+
+        testUpdateClassAttribute('null', null, '');
+        testUpdateClassAttribute('undefined', undefined, '');
+        testUpdateClassAttribute('empty string', '', '');
+        testUpdateClassAttribute('class string', 'my-class', 'my-class');
```

Comment:
Frankly this doesn't make a lot of sense (you get `null` at first, and then `''` after an update), but I'm just trying to test the existing behavior.